taichi/program/program.cpp

```diff
@@ -88,6 +90,13 @@ Program::Program(Arch desired_arch)
   } else if (config.arch == Arch::opengl) {
     TI_ASSERT(opengl::initialize_opengl(config.use_gles));
     program_impl_ = std::make_unique<OpenglProgramImpl>(config);
+  } else if (config.arch == Arch::dx11) {
+    if (!directx11::is_dx_api_available()) {
+      TI_WARN("No DX API detected.");
```

Comment:
nit: Should change it to "DirectX 11 Unavailable, can not detect a supported device" or something like that. DX API sounds a bit vague.

Reply:
Done.